Code/6-wire-version/src/main.cpp

```diff
@@ -167,7 +213,110 @@ void sendMQTT(){
 
 
 
-void startOTA() { // Start the OTA service
+/**
+ * Start a Wi-Fi access point, and try to connect to some given access points.
+ * Then wait for either an AP or STA connection
+ */
+void startWiFi()
+{
+  //WiFi.mode(WIFI_STA);
+  WiFi.setAutoReconnect(true);
+  WiFi.persistent(true);
+  WiFi.hostname(String("layzspa-" + ESP.getChipId()).c_str());
+
+  if (enableStaticIp4)
+  {
+    Serial.println("WiFi > using static IP \"" + ip4Address.toString() + "\" on gateway \"" + ip4Gateway.toString() + "\"");
+    WiFi.config(ip4Address, ip4Gateway, ip4Subnet, ip4DnsPrimary, ip4DnsSecondary);
+  }
+
+  if (enableAp)
+  {
+    Serial.println("WiFi > using WiFi configuration with SSID \"" + apSsid + "\"");
+    WiFi.begin(apSsid, apPwd);
+
+    Serial.print("WiFi > Trying to connect ...");
+    int maxTries = 5;
+    int tryCount = 0;
+
+    while (WiFi.status() != WL_CONNECTED)
+    {
+      delay(1000);
+      Serial.print(".");
+      tryCount++;
+
+      if (tryCount >= maxTries)
+      {
+        Serial.println("");
+        Serial.println("WiFi > NOT Connected!");
+        if (enableWmAp)
+        {
+          // disable specific WiFi config
+          enableAp = false;
+          enableStaticIp4 = false;
+          // fallback to WiFi config portal
+          startWiFiConfigPortal();
+        }
+        break;
+      }
+    }
+  }
+  else if (enableWmAp)
+  {
+    startWiFiConfigPortal();
+  }
+
+  if (WiFi.status() == WL_CONNECTED)
```

Comment:
WiFi is not ready yet at this point (in my setup). It will reconnect automatically later. But this code is not guaranteed to run. Possible solutions: Either put a delay before the test (which I don't like) or make a run once check some time after start-up.

Reply:
in my code it should be ready. the startWiFiConfigPortal() has an endless loop in its function to get the AP running.. think there should not be problem except a hardware failure - so its not a problem.
the static config is trying 5 times before aborting - 5 seconds - and then fallback to AP mode or just go ahead without wifi
Serial.println("WiFi > Connection failed. Retrying in a while");